src/compiler/parser/index.js

```diff
@@ -331,8 +331,20 @@ export function processElement (element: ASTElement, options: CompilerOptions) {
 function processKey (el) {
   const exp = getBindingAttr(el, 'key')
   if (exp) {
-    if (process.env.NODE_ENV !== 'production' && el.tag === 'template') {
-      warn(`<template> cannot be keyed. Place the key on real elements instead.`)
+    if (process.env.NODE_ENV !== 'production') {
+      if (el.tag === 'template') {
+        warn(`<template> cannot be keyed. Place the key on real elements instead.`)
+      }
+      if (el.for) {
+        const iterator = el.iterator2 || el.iterator1
+        const parent = el.parent
+        if (iterator && iterator === exp && parent && parent.tag === 'transition-group') {
+          warn(
+            `Do not use sequential index on the child of the <transtion-group> component, ` +
+            `this may cause some rendering errors.`
```

Comment:
I think we could say that this is the same as not using a `key` attribute and link to the same link the message _key is required in transition-group_ shows

Otherwise looks good 🙂

Reply:
There is already a warning message in the `<transition-group>` component：
```js
// In src/platforms/web/runtime/components/transition-group.js
render () {
  // omit...
  warn(`<transition-group> children must be keyed: <${name}>`)
  // omit...
}
```
😀❓❓❓

Reply:
@HcySunYang

I think @posva means that we can state “using index as `key`s is the same as not using a `key` at all” here.

@posva

`<transition-group> children must be keyed` currently links to nowhere...

Reply:
oh, I thought there was a link to the documentation in that warning